server/neptune/gateway/event/pull_request_handler.go

```diff
@@ -20,84 +44,113 @@ type PullRequest struct {
+	return p.Scheduler.Schedule(ctx, func(ctx context.Context) error {
+		return p.handle(ctx, request, event)
+	})
 }
 
-func (p *AutoplannerValidatorProxy) handle(ctx context.Context, request *http.BufferedRequest, event PullRequest) error {
-	if ok := p.autoplanValidator.InstrumentedIsValid(
-		ctx,
-		p.logger,
-		event.Pull.BaseRepo,
-		event.Pull.HeadRepo,
-		event.Pull,
-		event.User); ok {
-		return p.workerProxy.Handle(ctx, request, event)
+func (p *ModifiedPullHandler) handle(ctx context.Context, request *http.BufferedRequest, event PullRequest) error {
+	criteria := requirement.Criteria{
+		User:              event.User,
+		Branch:            event.Pull.HeadBranch,
+		Repo:              event.Pull.HeadRepo,
+		OptionalPull:      &event.Pull,
+		InstallationToken: event.InstallationToken,
+	}
+	if err := p.RequirementChecker.Check(ctx, criteria); err != nil {
+		return errors.Wrap(err, "checking pr requirements")
 	}
 
-	p.logger.WarnContext(ctx, "request isn't valid and will not be proxied to sns")
+	commit := &config.RepoCommit{
+		Repo:          event.Pull.HeadRepo,
+		Branch:        event.Pull.HeadBranch,
+		Sha:           event.Pull.HeadCommit,
+		OptionalPRNum: event.Pull.Num,
+	}
 
-	return nil
-}
+	// set clone depth to 1 for repos with a branch checkout strategy
+	cloneDepth := 0
+	matchingRepo := p.GlobalCfg.MatchingRepo(event.Pull.HeadRepo.ID())
+	if matchingRepo != nil && matchingRepo.CheckoutStrategy == "branch" {
+		cloneDepth = 1
+	}
+	builderOptions := config.BuilderOptions{
+		RepoFetcherOptions: &github.RepoFetcherOptions{
+			CloneDepth: cloneDepth,
+		},
+	}
 
-func (p *AutoplannerValidatorProxy) Handle(ctx context.Context, request *http.BufferedRequest, event PullRequest) error {
-	return p.scheduler.Schedule(ctx, func(ctx context.Context) error {
-		return p.handle(ctx, request, event)
-	})
-}
+	rootCfgs, err := p.RootConfigBuilder.Build(ctx, commit, event.InstallationToken, builderOptions)
+	if err != nil {
+		return errors.Wrap(err, "generating roots")
+	}
 
-func NewPullEventWorkerProxy(
-	snsWriter Writer,
-	logger logging.Logger,
-) *PullEventWorkerProxy {
-	return &PullEventWorkerProxy{
-		snsWriter: snsWriter,
-		logger:    logger,
+	var legacyModeRoots []*valid.MergedProjectCfg
+	var platformModeRoots []*valid.MergedProjectCfg
+	for _, rootCfg := range rootCfgs {
+		if rootCfg.WorkflowMode == valid.PlatformWorkflowMode {
+			platformModeRoots = append(platformModeRoots, rootCfg)
+		} else {
+			legacyModeRoots = append(legacyModeRoots, rootCfg)
+		}
 	}
-}
 
-type PullEventWorkerProxy struct {
-	snsWriter Writer
-	logger    logging.Logger
+	// TODO: remove when we deprecate legacy mode
+	if err := p.handleLegacyMode(ctx, request, event, rootCfgs, legacyModeRoots); err != nil {
+		return errors.Wrap(err, "handling legacy mode")
+	}
+	if err := p.handlePlatformMode(ctx, event, platformModeRoots); err != nil {
+		return errors.Wrap(err, "handling platform mode")
+	}
+	return nil
```

Comment:
Would much rather you break these out into separate structs under the same interface, testing is going to become a nightmare and then ripping the code out will make ripping the tests out also a nightmare.

Reply:
Fwiw, I didn't do that because `handlePlatformMode` really is just going to run one line: `SignalWithStartWorkflow(platformModeRoots)` and all of the logic is just going to be delegated to the signaler code in a subsequent PR. If I understand your request, they shouldn't share interfaces since these handler functions have slightly different parameters. That said I can isolate out handleLegacyMode to a separate struct

Reply:
Sounds good